proposals/infra/INF-0005-Long-Vector-TestPlan.md

```diff
@@ -0,0 +1,243 @@
+'ExecTests' source code in the DXC repo. There is a script in the WinTools repo
+which generates and annotates the HLK tests.
+
+There are three test categories we are concerned with:
+
+1. Implement DXIL Intrinsic tests:
+    - At the bottom of this document there is a table of all HLSL intrinsics and
+    their mapped DXIL OpCodes. We just need coverage for each DXIL OpCode. If the
+    DXIL OpCode column lists 'emulated' then this means that there are multiple
+    LLVM/DXIL Ops used to compute the intrinsic. We will need to do an audit of
+    them to confirm coverage of all vector ops.
+
+2. Implement LLVM Native operation tests:
+     - These are the test cases for the 'basic' HLSL operators listed in the HLSL
+     operators table below.
+     - We have added some test cases for these with the long vectors work, but we
+     will want to make sure we add HLK coverage for all of these.
+
+3. Standard loading and storing of long vectors
+     - Ensure we have some basic tests doing standard loading/storing of long vectors.
+
+# Vector Sizes to test
+
+I don't think there are any particularly interesting vector sizes to test. So I
+propose testing sizes of 5, 25, 100, 500. Sizes < 5 are assumed to already be
+covered by existing test collateral. But, as part of this work we will verify that
+assumption.
+
+# Implementation phases
+Do the test work in two simple phases.
+
+1. Implement and validate (locally against WARP) the 3 test categories.
+2. HLK related work:
+     - Add a SM 6.9 HLK requirement. Includes updating the HLK requirements doc.
+     - Update mm_annotate_shader_op_arith_table.py to annotate the new test cases
+        with HLK GUIDS and requirements
+     - Add new tests to HLK playlist
+
+# Shipping
+1. Tests will be shared privately with IHV's along with the latest DXC and
+latest Agility SDK for tesing and validation.
+
+2. The tests will ship with the HLK at a TBD date in a later OS release.
+
+# New HLK Tests
+mm_annotate_shader_op_arith_table.py (WinTools repo) will need to be updated to
+recognize any new tests (not test cases) added. And additional GUIDs added for new
+test cases. mm_annotate_shader_op_arith_table.py is called by mm-hlk-update.py when
+converting from 'ExecTests' to the HLK 'DxilConf' tests. The aforementioned
+*table.py script is run by Integration.HLKTestsUpdate.yaml
+
+# Test Validation Requirements
+The following statements must be true and validated for this work to be considered
+completed.
+ - All new test cases pass when run locally against a WARP device
+ - All new test cases are confirmed to be lit up and available in the HLK when the
+    target device reports support
+ - All new tests/test cases are added to the HLK playlist
+
+# Requirements
+     - Greg's long vector changes. Check-in ETA of 3/7/25.
+     - WARP long vector support (Jesse). Needs Greg's work, or a private WIP branch
+        of Greg's work. ETA of ~1 week to implement.
```

Comment:
This is out of date now.  You should be able to work against microsoft/DirectXShaderCompiler#7188.